### PR TITLE
ledger: wallet types  + derivation helper

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-290

## What changed (plus any additional context for devs)

added ledger types for handling future work 
added helper method to handle wallet derivation paths based on index, hooked into existing fn's

edited how we get wallet address to fix some TS errors what were preexisting 

had to tweak ledger dep versions back to what is being used in the previous PRs: https://github.com/rainbow-me/rainbow/pull/4238/files

- running into some funky shit because of how ledger set up all their stuff and not worth spending time to patch packages, etc. inconsequential changes since

## What to test

importing seed phrase, pkey, read only

### CODE REVIEW

File: src/utils/wallet.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    walletType: WalletLibraryType.ledger,
+  };
+};
+
+export const deriveAccountFromMnemonic = async (
+  mnemonic: string,
+  index = 0
+) => {
+  let seed;
+  if (ios) {
```
Comment (benisgold) [inline src/utils/wallet.ts:51]: u should use the new env variable `IS_IOS`

**Review by benisgold — APPROVED** [review]

File: src/helpers/signingWallet.ts [hunk 2]
```diff
@@ -15,17 +15,18 @@ import {
...
   if (typeof alreadyExistingWallet !== 'string') {
     const walletSeed = generateMnemonic();
-    const {
-      wallet,
-      address,
-    } = await ethereumUtils.deriveAccountFromWalletInput(walletSeed);
+    const { wallet, address } = await deriveAccountFromWalletInput(walletSeed);
+
+    // @ts-ignore need to handle types in case wallet or address are null
+    if (!wallet || !address) return null;
```
Comment (estrattonbailey) [inline src/helpers/signingWallet.ts:29]: If if is ever actually going to happen we should have a `logger.error` here.

File: src/hooks/useImportingWallet.ts [hunk 3]
```diff
@@ -209,12 +210,12 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
...
-            const walletResult = await ethereumUtils.deriveAccountFromWalletInput(
-              input
-            );
+            const walletResult = await deriveAccountFromWalletInput(input);
             // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ address: string; isHDWallet: b... Remove this comment to see the full error message
             setCheckedWallet(walletResult);
-            const ens = await fetchReverseRecord(walletResult.address);
+            const ens = await fetchReverseRecord(
+              walletResult.address as string
+            );
```
Comment (estrattonbailey) [inline src/hooks/useImportingWallet.ts:218]: Another case where if it's possible to be undefined, might want to error

File: src/model/wallet.ts [hunk 4]
```diff
@@ -200,6 +205,21 @@ const authenticationPrompt = lang.t('wallet.authenticate.please');
...
+export const getHdPath = ({
+  type,
+  index,
+}: {
+  type: WalletLibraryType;
+  index: number;
+}): string => {
+  switch (type) {
+    case WalletLibraryType.ledger:
+      return `m/44'/60'/${index}'/0/0`;
```
Comment (estrattonbailey) [inline src/model/wallet.ts:217]: If you have a link to docs for this or something, mind throwing a `@see` comment at the top of this method?

**Review by estrattonbailey — APPROVED** [review]
LGTM, only wondering about the unhappy-paths and logging

File: src/hooks/useImportingWallet.ts [hunk 5]
```diff
@@ -30,8 +30,10 @@ import { walletInit } from '@/model/wallet';
 import { Navigation, useNavigation } from '@/navigation';
 import { walletsLoadState } from '@/redux/wallets';
 import Routes from '@/navigation/routesNames';
-import { ethereumUtils, sanitizeSeedPhrase } from '@/utils';
+import { sanitizeSeedPhrase } from '@/utils';
 import logger from '@/utils/logger';
+import { deriveAccountFromWalletInput } from '@/utils/wallet';
+import { logger as Logger, RainbowError } from '@/logger';
```
Comment (jinchung) [inline src/hooks/useImportingWallet.ts:36]: can move over usage from old logger (line 34)

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**